core/deployment/src/main/java/io/quarkus/deployment/pkg/NativeConfig.java

```diff
@@ -195,11 +195,27 @@ public boolean isContainerBuild() {
     }
 
     /**
-     * The docker image to use to do the image build
+     * The docker image to use to do the image build. It can be one of `graalvm`, `mandrel`, or the full image path, e.g.
+     * {@code quay.io/quarkus/ubi-quarkus-mandrel:21.3-java17}.
      */
     @ConfigItem(defaultValue = "${platform.quarkus.native.builder-image}")
     public String builderImage;
 
+    public String getBuilderImage() {
+        try {
+            switch (BuilderImageProvider.valueOf(this.builderImage.toUpperCase())) {
+                case GRAALVM:
+                    return "quay.io/quarkus/ubi-quarkus-native-image:21.3-java11";
+                case MANDREL:
+                    return "quay.io/quarkus/ubi-quarkus-mandrel:21.3-java11";
+                default:
+                    throw new IllegalStateException();
+            }
+        } catch (IllegalArgumentException e) {
+            return this.builderImage;
+        }
```

Comment:
Probably best to avoid using exceptions for control flow :)

Reply:
I don't like it either. The alternative would be to use constants instead of an enum, unless I am missing some pattern :)
What do you suggest?

Reply:
Can't we just check all the values of the enum if they match?